src/wallet/asyncrpcoperation_sendmany.h

```diff
@@ -25,14 +25,36 @@
 
 using namespace libzcash;
 
-// A recipient is a tuple of address, amount, memo (optional if zaddr)
-typedef std::tuple<std::string, CAmount, std::string> SendManyRecipient;
+class SendManyRecipient {
+public:
+    std::string address;
+    CAmount amount;
+    std::string memo;
 
-// Input UTXO is a tuple (quadruple) of txid, vout, amount, coinbase)
-typedef std::tuple<uint256, int, CAmount, bool> SendManyInputUTXO;
+    SendManyRecipient(std::string address_, CAmount amount_, std::string memo_ = "") :
```

Comment:
I don't think the optionality of the memo argument is ever used.
```suggestion
    SendManyRecipient(std::string address_, CAmount amount_, std::string memo_) :
```